internal/helper/credential.go

```diff
@@ -37,6 +37,10 @@ type GitCredential struct {
 	// appear in the HTTP response. This attribute is one-way from Git to pass additional information to credential
 	// helpers.
 	WwwAuth []string
+	// BearToken when set to true will set the 'authtype=bearer' in the credential helper
+	BearerToken bool
```

Comment:
I thought the scm service was telling you the auth type rather than a boolean

Reply:
Yes, and we set this field when the token type is set to [TOKEN_TYPE_BEARER](https://github.com/cloudbees-io/checkout/blob/b19082674ae018b5b90ba43c8f58e86d7dc39208/cmd/helper.go#L212) by SCM service. This is not a user configurable field.

Reply:
Replaced the bool with a field to explicitly define the `authtype` in the credential helper file.